hubgrep/lib/hosting_service_interfaces/_hosting_service_interface.py

```diff
@@ -123,3 +126,17 @@ def default_api_url_from_landingpage_url(landingpage_url: str) -> str:
     def normalize_url(url):
         response = requests.head(url)
         return response.url
+
+
+class HostingServiceInterfaceResult:
+    hosting_service_interface: HostingServiceInterface
+    response: CachedResponse
+    search_results: List[SearchResult]
+
+    def __init__(self,
+                 hosting_service_interface: HostingServiceInterface,
+                 response: CachedResponse,
+                 search_results: List[SearchResult]):
+        self.hosting_service = hosting_service_interface
+        self.response = response
+        self.search_results = search_results
```

Comment:
And this is then the new internal result class (internal as within fetching stuff before aggregating)